lib/waterline.js

```diff
@@ -711,7 +711,19 @@ function Waterline() {
         });//</ each model identity >
 
         // Call the `registerDatastore` adapter method.
-        datastore.adapter.registerDatastore(datastore.config, usedSchemas, next);
+        datastore.adapter.registerDatastore(datastore.config, usedSchemas, function registerDatastoreCb(err) {
+          if (err) {
+            return next(err);
+          }
+
+          // Check if the datastore adapter has a `validateConnection` method
+          if (_.has(datastore.adapter, 'validateConnection')) {
```

Comment:
Could you instead use `getConnection` and `releaseConnection` adapter methods (you'll still need to check if they both exist)
Heres an example of how you can access the `manager`
https://github.com/balderdashy/sails-hook-orm/blob/9c2a3c768eba152e47af645d02462ed831f87dbb/lib/build-registered-datastore-instance.js#L116

Reply:
@eashaw The idea here was to abstract the details of things such as `getConnection` and `releaseConnection` (which are more common in the SQL realm and might not be applicable in other cases) away from waterline and keep it contained within the adapters (as done in https://github.com/balderdashy/sails-postgresql/pull/290). This way each adapter can be more flexible in terms of how it wants to validate a connection and waterline needn't be concerned with the implementation details of how connectivity to a datastore is to be validated. This is my opinion though, Let me know if you still want me to change this.